From 3fcef0a7a89b89f35bcc3b5cbd34f9fdc06aa019 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Sun, 2 Jan 2022 11:04:46 +0530
Subject: [PATCH 0124/1141] dt-bindings: usb: add STM32 dwc3 DRD ctrl

DT bindings yaml file for STM32 dwc3 glue layer to manage dwc3 DRD ctrl

Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Change-Id: I36954e44997179113fac7de88525539f9364374d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/305596
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 .../devicetree/bindings/usb/stm32,dwc3.yaml   | 103 ++++++++++++++++++
 1 file changed, 103 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/usb/stm32,dwc3.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/usb/stm32,dwc3.yaml
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/usb/stm32,dwc3.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STMicroelectronics STM32 MPU DWC3 USB SoC controller
+
+description:
+  The STM32 MPU DWC3 USB SoC controller block supports both Host(SS/HS/FS/LS)
+  or Device(SS/HS/FS) mode operation
+
+maintainers:
+  - Fabrice Gasnier <fabrice.gasnier@foss.st.com>
+
+properties:
+  compatible:
+    const: st,stm32mp25-dwc3
+
+  "#address-cells":
+    const: 1
+
+  "#size-cells":
+    const: 1
+
+  ranges: true
+
+  st,syscfg:
+    $ref: /schemas/types.yaml#/definitions/phandle-array #(arm/stm32/st,stm32-syscon.yaml#)
+    description: <Phandle to the STM32-MPU System Controller DT node> <offset-for-usr3drcr-reg>
+
+  st,vbusen-active-low:
+    type: boolean
+    description: VBUS-ENABLE signal polarity is active-low
+
+  st,ovrcur-active-low:
+    type: boolean
+    description: Over-Current signal polarity is active-low
+
+  st,enable-port-power-control:
+    type: boolean
+    description: Enable Host-Mode Port Power Control (bit-3 of capability param HCCPARAMS)
+
+  wakeup-source: true
+
+  interrupts:
+    maxItems: 1
+    description: interrupt used for wakeup
+
+# Required child node:
+
+patternProperties:
+  "^usb@[0-9a-f]+$":
+    $ref: snps,dwc3.yaml#
+
+required:
+  - compatible
+  - "#address-cells"
+  - "#size-cells"
+  - ranges
+  - st,syscfg
+
+allOf:
+  - if:
+      required:
+        - wakeup-source
+    then:
+      anyOf:
+        - required: [interrupts]
+        - required: [interrupts-extended]
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/stm32mp25-clks.h>
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+    #include <dt-bindings/reset/stm32mp25-resets.h>
+
+    usb3dr: usb@48300000 {
+        compatible      = "st,stm32mp25-dwc3";
+        st,syscfg       = <&syscfg 0x4800>;
+        #address-cells  = <1>;
+        #size-cells     = <1>;
+        ranges = <0x48300000 0x48300000 0x100000>;
+        st,ovrcur-active-low;
+        st,vbusen-active-low;
+        wakeup-source;
+        interrupts-extended = <&exti1 44 IRQ_TYPE_EDGE_RISING>;
+
+        dwc3: usb@48300000 {
+            compatible      = "snps,dwc3";
+            reg             = <0x48300000 0x100000>;
+            interrupts      = <GIC_SPI 228 IRQ_TYPE_LEVEL_HIGH>;
+            clock-names     = "ref", "bus_early", "suspend";
+            clocks          = <&rcc CK_KER_USB2PHY2>, <&rcc CK_BUS_USB3DR>,
+                              <&rcc CK_KER_USB2PHY2>;
+            resets          = <&rcc USB3DR_R>;
+            phys            = <&usb2_phy>, <&usb3_phy>;
+            phy-names       = "usb2-phy", "usb3-phy";
+            wakeup-source;
+        };
+    };
